### PR TITLE
sql: use pcx wall_time in timezone() to avoid Utc::now

### PR DESCRIPTION
**Pull request by mjibson**

### CODE REVIEW

File: src/sql/src/func.rs [hunk 1]
```diff
@@ -1693,7 +1694,10 @@ lazy_static! {
                 params!(String, Timestamp) => BinaryFunc::TimezoneTimestamp, 2069;
                 params!(String, TimestampTz) => BinaryFunc::TimezoneTimestampTz, 1159;
                 // PG defines this as `text timetz`
-                params!(String, Time) => BinaryFunc::TimezoneTime, 2037;
+                params!(String, Time) => Operation::binary(|ecx, lhs, rhs| {
+                    let wall_time = DateTime::<Utc>::from(ecx.qcx.scx.pcx.wall_time).naive_utc();
+                    Ok(lhs.call_binary(rhs, BinaryFunc::TimezoneTime{wall_time}))
```
Comment (mjibson) [inline src/sql/src/func.rs:1699]: Is this the canonical way to do this? I'm not super clear on how this works and I just cargo-culted stuff until it compiled.

Comment (benesch) [reply src/sql/src/func.rs:1699]: Yeah seems good to me.

**Review by benesch — APPROVED** [review]

**Comment by mjibson** [comment]
Hrm, this change is now backward compatible with any views that used `timezone()` because the wall time is now fetched from the oneshot lifetime instead of SystemTime::now. Do we need to care about this, maybe just add a release note?

**Comment by benesch** [comment]
Eh yeah I think just a relnote is fine.

**Pull request merged**